File: src/Core/LotMatching/Schemas/Lot.py
from dataclasses import dataclass

from arrow import Arrow

from Core.LotMatching.Schemas.Trade import Trade


@dataclass
class LotAcquired:
    Date: Arrow
    Relation: Trade


@dataclass
class LotSold:
    Date: Arrow
    Relation: Trade


@dataclass
class Lot:
    Quantity: float
    Acquired: LotAcquired
    Sold: LotSold

File: lib/Core/LotMatching/Lot.ts
import type { ValidDateTime } from "@brrr/Utils/DateTime.ts";
import type { Trade } from "@brrr/Core/LotMatching/Trade.ts";

export type LotAcquired = {
	date: ValidDateTime;
	relation: Trade;
};

export type LotSold = {
	date: ValidDateTime;
	relation: Trade;
};

export type Lot = {
	quantity: number;
	acquired: LotAcquired;
	sold: LotSold;
};
